contract/scripts/deploy.ts

```diff
@@ -1,32 +1,30 @@
-import { ethers } from 'hardhat'
+import { ethers, upgrades } from 'hardhat'
 
 async function main() {
-    const [deployer] = await ethers.getSigners()
+    // ensure we have a server address, so web-server can authenticate with contract
+    const serverAddress = process.env.SERVER_ADDRESS
+    if (!serverAddress) {
+        console.error(
+            'No server address set in .env. Please set one manually or minting attempts will fail.'
+        )
+        return
+    }
+    console.log('Server address is:', serverAddress)
 
+    // deploy upgradeable contract
+    const [deployer] = await ethers.getSigners()
     console.log('Deploying contracts with the account:', deployer.address)
-
     console.log('Account balance:', (await deployer.getBalance()).toString())
 
-    const contract = await ethers.getContractFactory('PixelAvatars')
-    const token = await contract.deploy()
+    const Contract = await ethers.getContractFactory('PixelAvatars')
+    const token = await upgrades.deployProxy(Contract)
+    await token.setServerAddress(serverAddress)
 
-    console.log('Token address:', token.address)
-
-    const serverAddress = process.env.SERVER_ADDRESS
-
-    if (serverAddress) {
-        await token.setServerAddress(serverAddress)
-
-        console.log('Successfully updated server address to:', serverAddress)
-    } else {
-        console.warn(
-            'WARNING: No server address set in .env. Please set one manually or minting attempts will fail.'
-        )
-    }
+    console.log(
+        `OpenZeppelin Proxy deployed to: ${token.address} for VUE_APP_PIXEL_AVATAR_TOKEN and UPGRADEABLE_PROXY_ADDRESS`
```

Comment:
Perhaps we could make this even easier with a output like:

```
OpenZeppelin Proxy deployed to: 0xabc123...
> Please add the following line to your contract/.env: UPGRADEABLE_PROXY_ADDRESS=0xabc123...
> Please add the following line to your web-client/.env: VUE_APP_PIXEL_AVATAR_TOKEN=0xabc123...
```

Reply:
I like your new helpful output suggestion -- future "us" will appreciate that too.